cluster-operator/controllers/vmc/rancher.go

```
@@ -384,7 +417,22 @@ func getRegistrationYAMLFromRancher(rc *rancherutil.RancherConfig, rancherCluste
 		return "", err
 	}
 
-	return manifestContent, nil
+	data, err := httputil.ExtractFieldFromResponseBodyOrReturnError(manifestContent, "data", "unable to find data token in Rancher response")
+	if err != nil {
+		return "", err
+	}
+
+	var items []ClusterRegistrationTokens
+	json.Unmarshal([]byte(data), &items)
+	for _, item := range items {
+		if item.ClusterID == rancherClusterID && item.State == "active" {
+			log.Oncef("ClusterRegistrationToken exists for the cluster %s", rancherClusterID)
+			return item.Token, nil
+		}
+	}
+
+	log.Infof("No active clusterRegistrationToken found")
+	return "", nil
```

Should this be an error?

---

In this case, the request to lookup tokens succeeded but we did not find an existing, active reg token for the cluster. We then want to create a reg token. I suppose we could treat this as an error, but then the calling function would just always ignore any errors and proceed to creating a token. I guess there are arguments for both.